out_writeapi.go

```diff
@@ -208,6 +215,32 @@ func getConfigField[T int | bool](plugin unsafe.Pointer, key string, defaultval
 	return finval, nil
 }
 
+// this function creates a new managed stream based on the config struct fields
+func buildStream(ctx context.Context, config **outputConfig) error {
+	currManagedStream, err := (*config).client.NewManagedStream(ctx,
+		managedwriter.WithType((*config).streamType),
+		managedwriter.WithDestinationTable((*config).tableRef),
+		//use the descriptor proto when creating the new managed stream
+		managedwriter.WithSchemaDescriptor((*config).schemaDesc),
+		managedwriter.EnableWriteRetries((*config).retryBool),
+		managedwriter.WithMaxInflightBytes((*config).queueBytes),
+		managedwriter.WithMaxInflightRequests((*config).queueRequests),
+	)
+	(*config).managedStream = currManagedStream
```

Comment:
Should we set this only if the call succeeded?

Reply:
Done